homeassistant/components/homekit/type_sensors.py

```diff
@@ -128,21 +153,244 @@ def __init__(self, *args):
+            air_quality = density_to_air_quality(state)
+            pm_density = state
+
+            self.char_air_quality.set_value(air_quality)
+            self.char_particulate_density.set_value(pm_density)
+            _LOGGER.debug("%s: Set air_quality to %d", self.entity_id, air_quality)
+            _LOGGER.debug(
+                "%s: Set particulate_density to %d", self.entity_id, pm_density
+            )
+        else:
+            if state is not None:
+                self.char_air_quality.set_value(state)
+                _LOGGER.debug("%s: Set air_quality to %d", self.entity_id, state)
+
+            if nitrogen_dioxide_density is not None:
+                self.char_nitrogen_dioxide_density.set_value(nitrogen_dioxide_density)
+                _LOGGER.debug(
+                    "%s: Set nitrogen_dioxide_density to %d",
+                    self.entity_id,
+                    nitrogen_dioxide_density,
+                )
+
+            if voc_density is not None:
+                self.char_voc_density.set_value(voc_density)
+                _LOGGER.debug("%s: Set voc_density to %d", self.entity_id, voc_density)
+
+            if pm_2_5_density is not None:
+                if hasattr(self, "char_pm_2_5_density"):
+                    self.char_pm_2_5_density.set_value(pm_2_5_density)
+                    _LOGGER.debug(
+                        "%s: Set pm_2_5_density to %d", self.entity_id, pm_2_5_density
+                    )
+                else:
+                    _LOGGER.error(
+                        "%s: Attempted to update %s but does not have the required characteristic",
+                        self.entity_id,
+                        CHAR_PM_2_5_DENSITY,
+                    )
+
+            if pm_10_density is not None:
+                if hasattr(self, "char_pm_10_density"):
+                    self.char_pm_10_density.set_value(pm_10_density)
+                    _LOGGER.debug(
+                        "%s: Set pm_10_density to %d", self.entity_id, pm_10_density
+                    )
+                else:
+                    _LOGGER.error(
+                        "%s: Attempted to update %s but does not have the required characteristic",
+                        self.entity_id,
+                        CHAR_PM_10_DENSITY,
+                    )
+
+            if pm_density is not None:
+                if hasattr(self, "char_pm_density"):
+                    self.char_particulate_density.set_value(pm_density)
+                    _LOGGER.debug(
+                        "%s: Set particulate_density to %d", self.entity_id, pm_density
+                    )
+                else:
+                    _LOGGER.error(
+                        "%s: Attempted to update %s but does not have the required characteristic",
+                        self.entity_id,
+                        CHAR_AIR_PARTICULATE_DENSITY,
+                    )
+
+            """convert the given PM size to value HAP-NodeJS expects
```

Comment:
This comment is a bit confusing because we don't use HAP-NodeJS? Is there a similar reference in HAP-python

Reply:
yeah. in my codeing induced haze i was getting confused between HAP-NodeJS and HAP-python. in any case i updated to just reference the Apple docs directly.